forc/src/utils/defaults.rs

```diff
@@ -49,26 +49,64 @@ path = "tests/harness.rs"
 }
 
 pub(crate) fn default_program() -> String {
-    r#"script;
+    r#"contract ;
 
-fn main() {
+    abi MyContract {
 
-}
-"#
-    .into()
-}
+    }
 
-pub(crate) fn default_test_program() -> String {
-    r#"
+    impl MyContract for Contract {
 
-#[tokio::test]
-async fn harness() {
-    assert_eq!(true, true);
-}
+    }
 "#
     .into()
 }
 
+pub(crate) fn default_test_program(project_name: &str) -> String {
+    format!(
+        "{}{}{}{}{}",
+        r#"
+    use fuel_tx::{ContractId, Salt};
+    use fuels_abigen_macro::abigen;
+    use fuels_contract::{contract::Contract, parameters::TxParameters};
+    use fuels_signers::util::test_helpers;
+    
+    // Load abi from json
+    abigen!(MyContract, "./"#,
+        project_name,
+        r#".json");
+    
+    async fn get_contract_instance() -> (MyContract, ContractId) {
+    
+        // Build the contract
+        let salt = Salt::from([0u8; 32]);
+        let compiled = Contract::load_sway_contract("./out/debug/"#,
+        project_name,
+        r#".bin", salt).unwrap();
+    
+        // Launch a local network and deploy the contract
+        let (provider, wallet) = test_helpers::setup_test_provider_and_wallet().await;
+    
+    
+        let id = Contract::deploy(&compiled, &provider, &wallet, TxParameters::default()).await.unwrap();
+    
+        let instance = MyContract::new(id.to_string(), provider, wallet);
+    
+        (instance, id)
+    
+    }
+    
+    
+    #[tokio::test]
+    async fn can_get_contract_id() {
+    
+        let (_instance, _id) = get_contract_instance().await;
+    
+        // Now you have an instance of your contract you can use to test each functioncarg
```

Comment:
Might be good to link to The Fuels-rs Book once it's ready here

Reply:
Yeah that would definitely be helpful for out-of-the-box experience, do you think adding a TODO above the template generating function would suffice? Or just a link the the home page of the Fuel Book for now?

Reply:
Adding a TODO should be sufficient IMO